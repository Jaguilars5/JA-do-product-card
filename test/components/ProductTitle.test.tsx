import React from 'react';
import renderer from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductTitle', () => {
  test('Debe mostrar el componente correctamente co el titulo personalizado', () => {
    const wrapper = renderer.create(<ProductTitle title="Custom Product" />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('Debe de mostrar el componente con el nombre del producto', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductTitle />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
