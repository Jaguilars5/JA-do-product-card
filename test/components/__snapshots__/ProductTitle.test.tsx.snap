// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Debe de mostrar el componente con el nombre del producto 1`] = `
<div
  className=" productCard undefined "
>
  <span
    className="productDescription undefined"
  >
    Coffee Con Imagen
  </span>
</div>
`;

exports[`ProductTitle Debe mostrar el componente correctamente co el titulo personalizado 1`] = `
<span
  className="productDescription undefined"
>
  Custom Product
</span>
`;
